### PR TITLE
Make setDefault compatible with object type argument

### PR DESCRIPTION
**Pull request by stubailo**
Rebased version of https://github.com/meteor/meteor/pull/6997

### CODE REVIEW

File: packages/reactive-dict/reactive-dict.js [hunk 1]
```diff
@@ -79,8 +79,18 @@ _.extend(ReactiveDict.prototype, {
     }
   },
 
-  setDefault: function (key, value) {
+  setDefault: function (keyOrObject, value) {
     var self = this;
+
+    if ((typeof keyOrObject === 'object') && (value === undefined)) {
```
Comment (benjamn) [inline packages/reactive-dict/reactive-dict.js]: These extra parentheses aren't necessary, fwiw.

File: packages/reactive-dict/reactive-dict.js [hunk 2]
```diff
@@ -198,6 +208,14 @@ _.extend(ReactiveDict.prototype, {
     });
   },
 
+  _setDefaultObject: function (object) {
```
Comment (benjamn) [inline packages/reactive-dict/reactive-dict.js]: Why make this `_private`?

**Comment by tmeasday** [comment]
@stubailo are we now good to merge this? It passes!

**Comment by stubailo** [comment]
Doneee

**Pull request merged**